certificates.c

```diff
@@ -67,6 +67,49 @@ hibacert_from_ext(struct hibacert *cert, struct hibaext *ext,
 	return HIBA_OK;
 }
 
+int
+hibacert_load_extensions(struct hibacert *cert, struct sshbuf *extensions) {
+	int pos = 0;
+	int ret = 0;
+	size_t i = 0;
+	size_t len = 0;
+	const char *data = NULL;
+
+	if ((ret = sshbuf_get_string_direct(extensions, (const unsigned char**)&data, &len)) < 0) {
+		debug3("hibacert_load_extensions: sshbuf_get_string_direct returned %d: %s", ret, ssh_err(ret));
+		return HIBA_INTERNAL_ERROR;
+	}
+	debug3("hibacert_load_extensions: total extension len %ld", len);
+
+	// Multiple grants can be concatenated using ';'
+	do {
+		int extension_len = 0;
+		if (data[i] == ';') {
+			extension_len = i - pos;
+		}
+		if (i == len - 1) {
```

Comment:
Would the extension silently corrupt If I were to accidentally add a trailing semicolon? (Because in that case, we would enter both conditional blocks here.)

Reply:
It would not silently corrupt, but would fail explicitly parsing. Using a else if solves this issue.